tls/s2n_handshake.h

```diff
@@ -27,6 +27,25 @@
 #include "crypto/s2n_certificate.h"
 #include "crypto/s2n_hash.h"
 
+/* From RFC 8446: https://tools.ietf.org/html/rfc8446#appendix-B.3 */
+#define TLS_HELLO_REQUEST              0
+#define TLS_CLIENT_HELLO               1
+#define TLS_SERVER_HELLO               2
+#define TLS_SERVER_NEW_SESSION_TICKET  4
+#define TLS_ENCRYPTED_EXTENSIONS       8
+#define TLS_CERTIFICATE               11
+#define TLS_SERVER_KEY                12
+#define TLS_CERT_REQ                  13
+#define TLS_SERVER_HELLO_DONE         14
+#define TLS_CERT_VERIFY               15
+#define TLS_CLIENT_KEY                16
+#define TLS_FINISHED                  20
+#define TLS_SERVER_CERT_STATUS        22
+#define TLS_SERVER_SESSION_LOOKUP     23
+#define TLS_MESSAGE_HASH             254
+
+#define MESSAGE_HASH_HEADER_LENGTH  4
```

Comment:
Does this belong here, or in a new header file for s2n_transcript_hash.c?

Reply:
`TLS_MESSAGE_HASH` belongs with the rest of the TLS header messages. Since `MESSAGE_HASH_HEADER_LENGTH` is tightly coupled, I put it here as well. Since two units use the TLS header messages (s2n_handshake_io and s2n_handshake_transcript) I included all the TLS header messages in the s2n_handshake.h file.

I'm not sure why this exposes them any more than exposing the handshake struct, or the message_type_t enum.

Reply:
I just meant MESSAGE_HASH_HEADER_LENGTH, but github is enthusiastic with its highlight :) How is MESSAGE_HASH_HEADER_LENGTH tightly coupled to TLS_MESSAGE_HASH? Is it used outside of s2n_handshake_transcript?

If we're going to keep it in this header, can we preface it with S2N_?

Reply:
Updated